src/en/mangasaki/src/eu/kanade/tachiyomi/extension/en/mangasaki/MangaSaki.kt

```diff
@@ -0,0 +1,265 @@
+            pageList.add(Page(pageNumber, "", it))
+            pageNumber++
+        }
+
+        return pageList.sortedBy { page -> page.index }
+    }
+
+    override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException("Not used")
+
+    override fun getFilterList() = FilterList(
+        Filter.Header("NOTE: Ignored if using text search!"),
+        Filter.Separator(),
+        MangaSaki.GenreFilter(),
+    )
+
+    private class GenreFilter : MangaSaki.UriPartFilter(
+        "Category",
+        arrayOf(
+            Pair("Action", "action"),
+            Pair("Adult", "adult"),
+            Pair("Adventure", "adventure"),
+            Pair("Comedy", "comedy"),
+            Pair("Crime", "crime"),
+            Pair("Drama", "drama"),
+            Pair("Dungeons", "dungeons"),
+            Pair("Ecchi", "ecchi"),
+            Pair("Fantasy", "fantasy"),
+            Pair("GenderBender", "genderbender"),
+            Pair("Gender Bender", "gender-bender"),
+            Pair("Harem", "harem"),
+            Pair("Hentai", "hentai"),
+            Pair("Historical", "historical"),
+            Pair("Horror", "horror"),
+            Pair("Isekai", "isekai"),
+            Pair("Josei", "josei"),
+            Pair("Lolicon", "lolicon"),
+            Pair("Magical Girls", "magical-girls"),
+            Pair("MartialArts", "martialarts"),
+            Pair("Martial Arts", "martial-arts"),
+            Pair("Mature", "mature"),
+            Pair("Mecha", "mecha"),
+            Pair("Medical", "medical"),
+            Pair("N/A", "na"),
+            Pair("Philosophical", "philosophical"),
+            Pair("Psychological", "psychological"),
+            Pair("SchoolLife", "schoollife"),
+            Pair("School Life", "school-life"),
+            Pair("Sci-fi", "sci-fi"),
+            Pair("Sci-fi Shounen", "sci-fi-shounen"),
+            Pair("Seinen", "seinen"),
+            Pair("Shotacon", "shotacon"),
+            Pair("Shoujo", "shoujo"),
+            Pair("ShoujoAi", "shoujoai"),
+            Pair("Shoujo Ai", "shoujo-ai"),
+            Pair("Shounen", "shounen"),
+            Pair("ShounenAi", "shounenai"),
+            Pair("Shounen-Ai", "shounen-ai"),
+            Pair("SliceofLife", "slicelife"),
+            Pair("Slice of Life", "slice-life"),
+            Pair("Smut", "smut"),
+            Pair("Sports", "sports"),
+            Pair("Superhero", "superhero"),
+            Pair("Supernatural", "supernatural"),
+            Pair("System", "system"),
+            Pair("Thriller", "thriller"),
+            Pair("Tragedy", "tragedy"),
+            Pair("Webtoons", "webtoons"),
+            Pair("Wuxia", "wuxia"),
+            Pair("Yuri", "yuri"),
+        ),
+    )
+
+    private fun parseJSON(jsonString: String): List<String>? {
+        val jsonData = Json { ignoreUnknownKeys = true }.decodeFromString<JSONData>(jsonString)
+        return jsonData.showmanga.paths.filter { it.contains("mangasaki") }
+    }
```

Comment:
```suggestion
    private val json: Json by injectLazy()

    private fun parseJSON(jsonString: String): List<String> {
        val jsonData = json.decodeFromString<JSONData>(jsonString)
        return jsonData.showmanga.paths.filter { it.contains("mangasaki") }
    }
```
The app has a [`Json` instance](https://github.com/mihonapp/mihon/blob/96c236e5c38248c875f2ac7596cd51845aa651ea/app/src/main/java/eu/kanade/tachiyomi/di/AppModule.kt#L93-L98) that extensions can use if there aren't any special requirements. Remember to import `uy.kohesive.injekt.injectLazy`.